pages/api/auth/[...nextauth].ts

```diff
@@ -84,10 +90,19 @@ const onSignIn: EventCallbacks["signIn"] = async (message) => {
   const trackData =
     (await trackRes.json()) as SpotifyApi.UsersTopTracksResponse;
 
-  return setDoc(doc(profilesCol, user.id), {
+  const docRef = doc(profilesCol, user.id);
+  if ((await getDoc(docRef)).exists()) {
+    return updateDoc(docRef, {
+      topArtists: artistData.items,
+      topTracks: trackData.items,
+      updatedAt: serverTimestamp(),
+    });
+  }
+  return setDoc(docRef, {
     topArtists: artistData.items,
     topTracks: trackData.items,
     updatedAt: serverTimestamp(),
+    savedItemIds: [],
```

Comment:
If you end up making `savedItemIds` optional, you can maybe get rid of this if statement and just add the `{ merge: true }` option.

https://firebase.google.com/docs/firestore/manage-data/add-data#set_a_document